src/en/snowmtl/src/eu/kanade/tachiyomi/extension/en/snowmtl/Snowmtl.kt

```diff
@@ -0,0 +1,480 @@
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.ResponseBody.Companion.toResponseBody
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import rx.Observable
+import java.io.ByteArrayOutputStream
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Locale
+import kotlin.math.absoluteValue
+
+class Snowmtl : ParsedHttpSource() {
+
+    override val name = "Snow Machine Translations"
+
+    override val baseUrl = "https://snowmtl.ru"
+
+    override val lang = "en"
+
+    override val supportsLatest = true
+
+    override val client = network.cloudflareClient.newBuilder()
+        .addInterceptor(::composedImageIntercept)
+        .build()
+
+    private val captions = mutableMapOf<String, PageMapping>()
+
+    // ============================== Popular ===============================
+
+    private val popularFilter = FilterList(SelectionList("", listOf(Option(value = "views", query = "sort_by"))))
+
+    override fun popularMangaRequest(page: Int) = searchMangaRequest(page, "", popularFilter)
+
+    override fun popularMangaSelector() = searchMangaSelector()
+
+    override fun popularMangaFromElement(element: Element) = searchMangaFromElement(element)
+
+    override fun popularMangaNextPageSelector() = searchMangaNextPageSelector()
+
+    // =============================== Latest ===============================
+
+    private val latestFilter = FilterList(SelectionList("", listOf(Option(value = "recent", query = "sort_by"))))
+
+    override fun latestUpdatesRequest(page: Int) = searchMangaRequest(page, "", latestFilter)
+
+    override fun latestUpdatesSelector() = searchMangaSelector()
+
+    override fun latestUpdatesFromElement(element: Element) = searchMangaFromElement(element)
+
+    override fun latestUpdatesNextPageSelector() = searchMangaNextPageSelector()
+
+    // =========================== Search ============================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$baseUrl/search".toHttpUrl().newBuilder()
+            .addQueryParameter("page", page.toString())
+
+        if (query.isNotBlank()) {
+            url.addQueryParameter("query", query)
+        }
+
+        filters.forEach { filter ->
+            when (filter) {
+                is SelectionList -> {
+                    val selected = filter.selected()
+                    if (selected.value.isBlank()) {
+                        return@forEach
+                    }
+                    url.addQueryParameter(selected.query, selected.value)
+                }
+                is GenreList -> {
+                    filter.state.filter(GenreCheckBox::state).forEach { genre ->
+                        url.addQueryParameter("genres", genre.id)
+                    }
+                }
+                else -> {}
+            }
+        }
+
+        return GET(url.build(), headers)
```

Comment:
Since extensions-lib 1.4 you should be able to remove `.build()`

```suggestion
        return GET(url, headers)
```

Reply:
```url``` is a builder, and the extensions-lib accept ```HttpUrl``` and ```String```:  [eu/kanade/tachiyomi/network/Requests](https://github.com/mihonapp/mihon/blob/addb4ae9ad5f9294c70bce8b5eebd806115158b2/core/common/src/main/kotlin/eu/kanade/tachiyomi/network/Requests.kt#L18)